packages/components/src/mobile/bottom-sheet-text-control/index.native.js

```diff
@@ -78,8 +82,7 @@ const BottomSheetTextControl = ( {
 					<TextInput
 						label={ label }
 						onChangeText={ ( text ) => onChangeText( text ) }
-						onChange={ onChange( value ) }
-						value={ value }
+						defaultValue={ value }
```

Comment:
Looks like that using `defaultValue` will make it unnecessary to keep the value in the state. I think we could simplify this component if we use the incoming prop `initialValue` for the `defaultValue` prop and for the `onChangeText` we could invoke the prop `onChange` directly.

**Changes:**
- `defaultValue={ value }` => `defaultValue={ initialValue }`
- Remove `const [ value, onChangeText ] = useState( initialValue );`
- `onChangeText={ ( text ) => onChangeText( text ) }` => `onChangeText={ ( text ) => onChange( text ) }`

wdyt?

Reply:
That works perfectly, thank you for the suggestion and clear explanation of why it works! I was also able to remove the `useEffect` hook with the necessary changes and have implemented that in https://github.com/WordPress/gutenberg/pull/33975/commits/ccbef5cba209116b7b95fe7ac518b81611b706de.